matrix/lambdas/daemons/notifications.py

```diff
@@ -0,0 +1,53 @@
+import os
+
+from matrix.common.aws.redshift_handler import RedshiftHandler
+from matrix.common.etl import run_etl, transform_bundle, finalizer_update
+from matrix.common.logging import Logging
+
+logger = Logging.get_logger(__name__)
+
+
+class NotificationsHandler:
+
+    def __init__(self, bundle_uuid, bundle_version, event_type):
+        logger.info(f"Running NotificationsHandler with parameters: {bundle_uuid}, {bundle_version}, {event_type}")
+        self.bundle_uuid = bundle_uuid
+        self.bundle_version = bundle_version
+        self.event_type = event_type
+
+        self.redshift = RedshiftHandler()
+
+    def run(self):
+        if self.event_type == 'CREATE':
+            self.update_bundle()
+        elif self.event_type == 'DELETE' or self.event_type == 'TOMBSTONE':
+            self.remove_bundle()
+
+    def update_bundle(self):
+        query = {
```

Comment:
Do we need to check if this is an existing bundle first and do a check on versions? Do we need to delete before inserting? Does this overwrite each time?

Reply:
Good catch. I didn't implement a versions check since we aren't expecting updates to come through the DSS, but it would be simple enough to implement if we were expecting for it to be used. Prefer not to if we didn't have to to avoid issuing an unnecessary query.

Nothing would happen if the bundle doesn't exist, but it would be cleaner to short circuit this if no bundle existed. I'll check the DSS API to see what's the quickest way to check for the existence of a bundle.

In `finalizer_update`, it runs through [this process](https://docs.aws.amazon.com/redshift/latest/dg/merge-replacing-existing-rows.html):
1. transforms the bundle into temporary tables
2. replace rows in real tables with rows in temporary tables

Yes, it will delete before inserting and overwrites conflicting primary keys every time.

Reply:
Makes sense, especially considering we are a long way away from updated bundles